pallets/passkey/src/lib.rs

```diff
@@ -350,3 +362,63 @@ where
 		)
 	}
 }
+
+/// Frame system related checks
+#[derive(Encode, Decode, Clone, TypeInfo)]
+#[scale_info(skip_type_params(T))]
+pub struct FrameSystemChecks<T: Config + Send + Sync>(pub T::AccountId, pub Call<T>);
+
+impl<T: Config + Send + Sync> FrameSystemChecks<T>
+where
+	<T as frame_system::Config>::RuntimeCall:
+		From<Call<T>> + Dispatchable<Info = DispatchInfo, PostInfo = PostDispatchInfo>,
+{
+	/// Validates the transaction fee paid with tokens.
+	pub fn pre_dispatch(&self) -> Result<(), TransactionValidityError> {
+		let info = &self.1.get_dispatch_info();
+		let len = self.0.using_encoded(|c| c.len());
```

Comment:
Shouldn't this be the encoded length of the call?

Reply:
you are correct, good catch